data_pipelines/rne/database/process_rne.py

```diff
@@ -0,0 +1,294 @@
+        "CREATE INDEX IF NOT EXISTS file_pp ON dirigeants_pp (file_name);",
+        "CREATE INDEX IF NOT EXISTS file_pm ON dirigeants_pm (file_name);",
+        """CREATE INDEX IF NOT EXISTS idx_pp_siren_file_name
+        ON dirigeants_pp (siren, file_name);""",
+        """CREATE INDEX IF NOT EXISTS idx_pm_siren_file_name
+        ON dirigeants_pm (siren, file_name);""",
+    ]
+
+    for statement in index_statements:
+        cursor.execute(statement)
+
+
+def inject_records_into_db(file_path, db_path, file_type):
+    dirigeants_pp, dirigeants_pm = [], []
+
+    with open(file_path, "r") as file:
+        logging.info(f"Processing stock file: {file_path}")
+        try:
+            if file_type == "stock":
+                json_data = file.read()
+                data = json.loads(json_data)
+                for record in data:
+                    list_dirigeants_pp, list_dirigeants_pm = extract_dirigeants_data(
+                        record, file_type
+                    )
+                    dirigeants_pp = dirigeants_pp + list_dirigeants_pp
+                    dirigeants_pm = dirigeants_pm + list_dirigeants_pm
+            elif file_type == "flux":
+                for line in file:
+                    data = json.loads(line)
+                    for record in data:
+                        (
+                            list_dirigeants_pp,
+                            list_dirigeants_pm,
+                        ) = extract_dirigeants_data(record, file_type)
+                        dirigeants_pp = dirigeants_pp + list_dirigeants_pp
+                        dirigeants_pm = dirigeants_pm + list_dirigeants_pm
+        except json.JSONDecodeError as e:
+            raise Exception(f"JSONDecodeError: {e} in file {file_path}")
+        insert_dirigeants_into_db(dirigeants_pp, dirigeants_pm, file_path, db_path)
+
+
+def find_and_delete_same_siren(cursor, siren, file_path):
+    """
+    Find and delete rows with the same SIREN and a different file
+    name from the database.
+
+    Args:
+        cursor: The database cursor.
+        siren (str): The SIREN to search for and delete.
+        file_path (str): The file path to filter the rows.
+
+    """
+    cursor.execute(
+        "SELECT COUNT(*) FROM dirigeants_pp WHERE siren = ? AND file_name != ?",
+        (siren, file_path),
+    )
+    count_already_existing_siren = cursor.fetchone()[0]
+    # If existing rows are found, delete them
+    if count_already_existing_siren is not None:
+        cursor.execute(
+            "DELETE FROM dirigeants_pp WHERE siren = ? AND file_name != ?",
+            (siren, file_path),
+        )
+
+
+def insert_dirigeants_into_db(
+    list_dirigeants_pp, list_dirigeants_pm, file_path, db_path
+):
+    connection, cursor = connect_to_db(db_path)
+
+    for dirigeant_pp in list_dirigeants_pp:
+        find_and_delete_same_siren(cursor, dirigeant_pp["siren"], file_path)
+        cursor.execute(
```

Comment:
You insert `dirigeant` one by one ? Would nt a bulk insert be faster ?

Reply:
Yes, it would. But a bulk delete will not allow for detecting the duplicates, discussed above. Since I use each `dirigeant`'s siren to detect duplicates.